### PR TITLE
Support attributes on primary constructors

### PR DESCRIPTION
**Pull request by AlekseyTs**
https://github.com/dotnet/csharplang/blob/main/proposals/primary-constructors.md#attributes-targeting-primary-constructors

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @jjonescz, @dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedPrimaryConstructor.cs [hunk 1]
```diff
@@ -99,5 +118,55 @@ public IEnumerable<FieldSymbol> GetBackingFields()
...
+            arguments.SymbolPart = AttributeLocation.Method;
+        }
+
+        internal override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> allAttributeSyntaxNodes, BindingDiagnosticBag diagnostics, AttributeLocation symbolPart, WellKnownAttributeData decodedData)
+        {
+            Debug.Assert(symbolPart is AttributeLocation.Method or AttributeLocation.Return);
+            base.PostDecodeWellKnownAttributes(boundAttributes, allAttributeSyntaxNodes, diagnostics, symbolPart is AttributeLocation.Method ? AttributeLocation.None : symbolPart, decodedData);
+        }
+
+        protected override bool ShouldBindAttributes(AttributeListSyntax attributeDeclarationSyntax, BindingDiagnosticBag diagnostics)
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedPrimaryConstructor.cs:145]: nit: It is a bit odd to report diagnostics from a method called "ShouldXYZ" (wouldn't expect to have such side-effects).

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs [hunk 2]
```diff
@@ -3064,6 +3064,347 @@ public static void Main()
...
+            Assert.True(c1.HasPrimaryConstructor);
+            Assert.Equal("A", c1.PrimaryConstructor.GetAttributes().Single().ToString());
+            Assert.True(c1.Constructors.Where(ctor => ctor != c1.PrimaryConstructor).All(ctor => ctor.GetAttributes().IsEmpty));
+
+            var c2 = (SourceNamedTypeSymbol)comp.GetTypeByMetadataName("C2");
+            Assert.Empty(c2.GetAttributes());
+            Assert.True(c2.HasPrimaryConstructor);
+            Assert.Equal("B", c2.PrimaryConstructor.GetAttributes().Single().ToString());
+            Assert.True(c2.Constructors.Where(ctor => ctor != c2.PrimaryConstructor).All(ctor => ctor.GetAttributes().IsEmpty));
+        }
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs:3406]: I assume it's covered already (and orthogonal to the attribute target) but consider also testing some attributes with `nameof(parameter)`

**Comment by AlekseyTs** [comment]
@jjonescz, @dotnet/roslyn-compiler For the second review.

**Review by jjonescz — APPROVED** [review]

**Pull request merged**